lib/src/main/java/com/auth0/jwt/JWTCreator.java

```diff
@@ -98,6 +98,24 @@ public Builder withHeader(Map<String, Object> headerClaims) {
             return this;
         }
 
+        /**
+         * Add specific Claims to set as the Header.
+         * If provided json is null then nothing is changed
+         *
+         * @param headerClaimsJson the values to use as Claims in the token's Header.
+         * @return this same Builder instance.
+         * @throws JsonProcessingException if json value has invalid structure
+         */
+        public Builder withHeader(String headerClaimsJson) throws JsonProcessingException {
```

Comment:
Instead of throwing an exception from jackson, let's catch and throw an `IllegalArgumentException` with the nested cause, to be consistent with the other methods and not leak the jackson dependency.